lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -49,6 +51,135 @@ export class TokenValidator {
         this.openIdConfigProvider = new OpenIdConfigProvider(this.config, this.networkInterface, this.logger);
     }
 
+    /**
+     * Middleware-style function to validate token from request.
+     * @param {@link (TokenValidationParameters:type)}
+     * @param resource Optional resource to retrieve access token from session 
+     * @returns 
+     */
+    validateTokenMiddleware(options: TokenValidationParameters, resource?: string): TokenValidationMiddlewareResponse {
+
+        // @ts-ignore 
+        return (req: ExpressRequest, res: ExpressResponse, next: ExpressNextFunction) => {
+
+            // Adding access token to authorization header from session
+            if (resource && req.session && req.session.protectedResources) {
+                if (!req.session.protectedResources[resource]) {
+                    const error = ValidationConfigurationError.createMissingTokenError(`Resource ${resource} is not a protectedResource in req session. No access token to add to headers.`);
+                    next(error);
+                } else {
+                    const token = req.session.protectedResources[resource].accessToken;
+                    if (!token) {
+                        const error = ValidationConfigurationError.createMissingTokenError(`No access token for resource ${resource} in session. Unable to add to request headers.`);
+                        next(error);
+                    }
+    
+                    // Attach bearer token to authorization header
+                    req.headers.authorization = `Bearer ${token}`;
```

Comment:
Are we only supporting Bearer right now?

Reply:
Yes, just bearer at the moment.